### PR TITLE
Put all apps in ./apps/, infrastructure in ./infrastructure/

### PR DESCRIPTION
**Pull request by mweiden**
This PR restructures the repo to have all applications in a single
directory called 'apps'. This solves two longstanding issues in the repo

1. The location of apps within the repo was overly fragmented
2. The location of apps within the repo was not adequately descriptive
   to merit complicating the directory structure

Further, this PR moves all infrastructure files to a directory called
`infrastructure`.

### CODE REVIEW

**Comment by mweiden** [comment]
@parthshahva 
screen shot 2018-03-19 at 6 43 13 pm

f0f8f5889fb6ff45e802cf33201177c5

File: apps/firehose_to_es_processor/main.tf [hunk 1]
```diff
@@ -16,6 +16,7 @@ terraform {
   backend "s3" {
     key = "logs/firehose-to-es-processor.tfstate"
     region = "us-east-1"
+    bucket = "org-humancellatlas-861229788715-terraform"
```
Comment (parthshahva) [inline apps/firehose_to_es_processor/main.tf]: Account id as above

File: apps/cwl_to_slack/main.tf [hunk 2]
```diff
@@ -4,12 +4,14 @@ variable "region" {
...
 provider "aws" {
   region = "${var.region}"
+  profile = "hca"
 }
 
 terraform {
   backend "s3" {
     key = "logs/cwl-to-slack-notifier.tfstate"
     region = "us-east-1"
+    bucket = "org-humancellatlas-861229788715-terraform"
```
Comment (parthshahva) [inline apps/cwl_to_slack/main.tf]: Account id should be hidden?

File: apps/cwl_firehose_subscriber/main.tf [hunk 3]
```diff
@@ -16,6 +16,7 @@ terraform {
   backend "s3" {
     key = "logs/cwl-firehose-subscriber.tfstate"
     region = "us-east-1"
+    bucket = "org-humancellatlas-861229788715-terraform"
```
Comment (parthshahva) [inline apps/cwl_firehose_subscriber/main.tf]: Account id should be hidden?

File: infrastructure/infrastructure.tf [hunk 4]
```diff
@@ -15,6 +16,7 @@ terraform {
   backend "s3" {
     key = "logs/terraform.tfstate"
     region = "us-east-1"
+    bucket = "org-humancellatlas-861229788715-terraform"
```
Comment (parthshahva) [inline infrastructure/infrastructure.tf]: Account id as above

**Pull request merged**